code/modules/clothing/modular_armor/attachments/modules.dm

```diff
@@ -247,16 +247,33 @@
 	icon_state = "mod_chemsystem"
 	item_state = "mod_chemsystem_a"
 	slot = ATTACHMENT_SLOT_MODULE
+	///Lets us keep track of what icon state we're in
+	var/chemsystem_is_active = FALSE
 
 /obj/item/armor_module/module/chemsystem/on_attach(obj/item/attaching_to, mob/user)
 	. = ..()
-	parent.AddComponent(/datum/component/chem_booster)
+	var/datum/component/chem_booster/chemsystem = parent.AddComponent(/datum/component/chem_booster)
```

Comment:
doesn't need to be casted really, nor needs to be a variable.

Your solution is fine, I am just commenting to make sure you fully understand what you are doing